api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -593,6 +588,23 @@ public void mergePatients(Patient preferred, Patient notPreferred) throws APIExc
 		Context.getPersonService().savePersonMergeLog(personMergeLog);
 	}
 	
+	private void requireNoActiveOrderOfSameType(Patient patient1, Patient patient2) {
+		String messageKey = "Patient.merge.cannotHaveSameTypeActiveOrders";
+		List<Order> ordersByPatient1 = Context.getOrderService().getAllOrdersByPatient(patient1);
+		List<Order> ordersByPatient2 = Context.getOrderService().getAllOrdersByPatient(patient2);
+		ordersByPatient1.forEach((Order order1) -> {
+			ordersByPatient2.forEach((Order order2) -> {
+				if (order1.isActive() && order2.isActive() && order1.getOrderType().equals(order2.getOrderType())) {
```

Comment:
There may be a gap between our understandings, since there was further discussion on the openmrs talk after the design call on 11th of Jan. Could you have a look at it once more?
The second condition of the three conditions, you are referring to, we **didn't agree** to do.